chain-abci/src/app/app_init.rs

```diff
@@ -248,13 +249,24 @@ fn get_voting_power(
     }
 }
 
-fn store_valid_genesis_state(genesis_state: &ChainNodeState, inittx: &mut DBTransaction) {
+fn store_valid_genesis_state(
+    genesis_state: &ChainNodeState,
+    inittx: &mut DBTransaction,
+    write_history_states: bool,
+) {
     let encoded = genesis_state.encode();
     inittx.put(COL_NODE_INFO, LAST_STATE_KEY, &encoded);
-    inittx.put(COL_EXTRA, b"init_chain_state", &encoded);
+    let encoded_height = i64::encode_var_vec(0);
+    inittx.put(COL_APP_HASHS, &encoded_height, &genesis_state.last_apphash);
```

Comment:
I guess this was there before, but not sure if historical app hashes (i..e mapping height->app hash) are needed for validation or only for client querying (as abci query can specify a block height to query against)

Reply:
This is only to keep current behavior, it's only used in the query. Because `COL_APP_STATES` is optional, we can't simply remove the `COL_APP_HASHS`.